pkg/utils/dataset.go

```diff
@@ -71,6 +77,25 @@ func GetAccessModesOfDataset(client client.Client, name, namespace string) (acce
 
 }
 
+func GetPVCStorageCapacityOfDataset(client client.Client, name, namespace string) (storageCapacity resource.Quantity, err error) {
+	dataset, err := GetDataset(client, name, namespace)
+	if err != nil {
+		return storageCapacity, fmt.Errorf("failed to get dataset %s/%s: %w", namespace, name, err)
+	}
+	annotations := dataset.GetObjectMeta().GetAnnotations()
+	if annotations == nil {
+		storageCapacity = resource.MustParse(DefaultStorageCapacity)
+		return
+	}
+	size := annotations[PVCStorageAnnotation]
+	if size == "" {
+		storageCapacity = resource.MustParse(DefaultStorageCapacity)
+		return
+	}
+	storageCapacity, err = resource.ParseQuantity(size)
```

Comment:
How about  handling the scenario where parsing the annotation value fails by returning the default storage capacity and logging the error?


```golang
storageCapacity, err = resource.ParseQuantity(size)
	if err != nil{
		log.Info("Failed to parse storage capacity '%s', using default '%s': %v\n", size, DefaultStorageCapacity, err)
		return resource.MustParse(DefaultStorageCapacity), nil
	}
	return 
```